File: src/IO/dislin-11.0/python/polar.py
#! /usr/bin/env python
import math
import dislin

n = 300
m = 10
f = 3.1415927/180.
x1 = range (n)
y1 = range (n)
x2 = range (m) 
y2 = range (m)
step = 360./(n-1)
for i in range (0,n):
  a = (i * step) * f
  y1[i] = a
  x1[i] = math.sin (5*a)

for i in range (0,m):
  x2[i] = i + 1
  y2[i] = i + 1

dislin.setpag ('da4p')
dislin.metafl ('cons')
dislin.disini ()
dislin.complx ()
dislin.pagera ()

dislin.titlin ('Polar Plots', 2)
dislin.ticks  (3, 'Y')
dislin.axends ('NOENDS', 'X')
dislin.labdig (-1, 'Y')
dislin.axslen (1000, 1000)
dislin.axsorg (1050, 900)

dislin.polar  (1.,0., 0.2, 0., 30.)
dislin.curve  (x1, y1, n)
dislin.htitle (50)
dislin.title  ()
dislin.endgrf ()

dislin.labdig (-1, 'X')
dislin.axsorg (1050, 2250)
dislin.labtyp ('VERT', 'Y')
dislin.polar  (10.,0.,2.,0.,30.)
dislin.barwth (-5.)
dislin.polcrv ('FBARS')
dislin.curve  (x2, y2, m)

dislin.disfin ()

File: src/IO/dislin-11.0/java/polar.java
import de.dislin.Dislin;

public class polar {
  public static void main (String args []) {
     int n = 300, m = 10, i;
     double a, f = 3.1415926/180., step = 360. / (n-1);

     float x1 [] = new float [n];
     float y1 [] = new float [n];
     float x2 [] = new float [m];
     float y2 [] = new float [m];

     for (i = 0; i < n; i++) {
        a = (i * step) * f;
        y1[i] = (float) a;
        x1[i] = (float) Math.sin (5 * a);
     }

     for (i = 0; i < m; i++) {
        x2[i] = i + 1;
        y2[i] = i + 1;
     }    
    
     Dislin.setpag ("da4p");
     Dislin.metafl ("cons");
     Dislin.disini ();
     Dislin.pagera ();
     Dislin.complx ();

     Dislin.titlin ("Polar Plots", 2);
     Dislin.ticks  (3, "Y");
     Dislin.axends ("NOENDS", "X");
     Dislin.labdig (-1, "Y");
     Dislin.axslen (1000, 1000);
     Dislin.axsorg (1050, 900);

     Dislin.polar  (1.f, 0.f, 0.2f, 0.f, 30.f);
     Dislin.curve  (x1, y1, n);
     Dislin.htitle (50);
     Dislin.title  ();
     Dislin.endgrf ();

     Dislin.labdig (-1, "X");
     Dislin.axsorg (1050, 2250);
     Dislin.labtyp ("VERT", "Y");
     Dislin.polar  (10.f, 0.f, 2.f, 0.f, 30.f);
     Dislin.barwth (-5.f);
     Dislin.polcrv ("FBARS");
     Dislin.curve  (x2, y2, m);

     Dislin.disfin ();
  }
}
